core/src/main/java/io/grpc/internal/CallCredentialsApplyingTransportFactory.java

```diff
@@ -66,6 +69,21 @@ public void close() {
   private class CallCredentialsApplyingTransport extends ForwardingConnectionClientTransport {
     private final ConnectionClientTransport delegate;
     private final String authority;
+    // Negative value means transport active, non-negative value indicates shutdown invoked.
+    private final AtomicInteger pendingApplier = new AtomicInteger(Integer.MIN_VALUE + 1);
+    private final Status shutdownStatus = Status.UNAVAILABLE;
```

Comment:
1. This should be static.
2. You should not use a Status without a description (`withDescription()`). At worst, the description just acts as a unique string that can let you find where the status was created
3. You should actually use the status passed to shutdown/shutdownNow to fail RPCs. There's a couple ways to do that that are almost the same. You may choose to have shutdownNow set the savedShutdownStatus if it is null.